src/extractive.py

```diff
@@ -953,32 +955,25 @@ def test_epoch_end(self, outputs):
         for name, value in rouge_scores_log.items():
             self.log(name, value, prog_bar=False)
 
-    def predict(self, input_text, raw_scores=False, num_summary_sentences=3):
-        """Summaries ``input_text`` using the model.
-
-        Args:
-            input_text (str): The text to be summarized.
-            raw_scores (bool, optional): Return a dictionary containing each sentence
-                and its corespoding score instead of the summary. Defaults to False.
-            num_summary_sentences (int, optional): The number of sentences in the
-                output summary. This value specifies the number of top sentences to
-                select as the summary. Defaults to 3.
-
-        Returns:
-            str: The summary text. If ``raw_scores`` is set then returns a dictionary
-            of input sentences and their corespoding scores.
-        """
-        nlp = English()
-        sentencizer = nlp.create_pipe("sentencizer")
-        nlp.add_pipe(sentencizer)
-        doc = nlp(input_text)
-
+    def predict_sentences(
+        self, input_sentences: Union[List[str], doc.Doc], raw_scores=False, num_summary_sentences=3, tokenized=False
```

Comment:
Nice! I think this should be Span instead of Doc

Reply:
`type(nlp("This is a test document. Test sentence.").sents)` is a `generator` so yes we need to check for a generator instead of a spacy `Doc`.